django_celery_beat/utils.py

```diff
@@ -24,7 +26,15 @@ def make_aware(value):
     else:
         # naive datetimes are assumed to be in local timezone.
         if timezone.is_naive(value):
-            value = timezone.make_aware(value, timezone.get_default_timezone())
+            tm_isdst = time.localtime().tm_isdst
+
+            # tm_isdst may return -1 if it cannot be determined
```

Comment:
I am not sure about this part, can any other way we can check it?

Reply:
I referenced these posts [1](https://stackoverflow.com/questions/2881025/python-daylight-savings-time) (top answer only) and [2](https://stackoverflow.com/questions/39610892/in-python-how-to-check-if-a-given-time-is-in-daylight-saving-time-period). There are other suggested solutions online. I can summarize the problems with these solutions.

### Use of `localize`
[Source 1](https://gist.github.com/dpapathanasiou/09bd2885813038d7d3eb)
[Source 2](https://stackoverflow.com/questions/2881025/python-daylight-savings-time)
[Source 3](https://stackoverflow.com/questions/31146092/how-to-determine-if-a-timezone-specific-date-in-the-past-is-daylight-saving-or-n)

The root cause was inside `localize`. Using the `localize` to fix the problem with `localize` sounds counterintuitive. For instance, link 1 mentions the following in the final solution:

```
Once you've **localized** a datetime in fact you can simply check it's .dst() method to see if it's zero.
```

### `time.daylight`
[Source](https://www.geeksforgeeks.org/python-time-daylight-function/)

This solution might work, but the underlying code is just to do a similar check to what I do here. This code is cleaner, and I can change it if you would like.

### Calculations
[Source](https://stackoverflow.com/questions/38722792/check-if-date-is-in-daylight-savings-time-for-timezone-without-pytz)

One way is to calculate the timezone based on the current time and a previous time, but I think that this solution is long and too ad-hoc when we can achieve this with a few lines of code.

### Conclusion
I think that `time.daylight` or `time.localtime`+`tm_isdst` is the best way to go for this. If you believe in the other solutions, please let me know.

Reply:
Sounds good. Let me know when you've checked.